validmind/tests/model_validation/sklearn/ClassifierPerformance.py

```diff
@@ -126,9 +126,18 @@ def run(self):
             output_dict=True,
             zero_division=0,
         )
-        report["roc_auc"] = multiclass_roc_auc_score(
-            self.inputs.dataset.y,
-            self.inputs.dataset.y_pred(self.inputs.model),
-        )
+
+        y_true = self.inputs.dataset.y
+
+        if len(unique(y_true)) > 2:
```

Comment:
```suggestion
        if len(np.unique(y_true)) > 2:
```

I think using `np.unique` here instead of `unique` might be a bit more performant especially for larger datasets but I am not 100% sure on that?

A better long-term solution might be to check the number target classes set on the dataset class: `len(self.inputs.dataset.target_class_labels)`. The problem there is its not guaranteed that the user passes the `class_labels` argument when running `init_dataset` so it could be wrong. I am thinking we update this line of the dataset class: https://github.com/validmind/developer-framework/blob/main/validmind/vm_models/dataset/dataset.py#L108 to be the following:

```python
        # if class labels not supplied, use unique values in target column
        self.target_class_labels = (
            target_class_labels
            if target_class_labels
            else {str(i): str(i) for i in np.unique(self._df[self.target_column])}
        )
```

What do you think?

Reply:
Yep, `np.unique(y_true)` works since we can't guarantee the user will pass `target_class_labels`

Reply:
I like the idea of setting and using the `target_class_labels`. I will use `np.unique(y_true)` for this one, but happy to introduce the idea of setting `target_class_labels` going forward.